packages/db/src/types.ts

```diff
@@ -35,6 +37,8 @@ export interface PostgresCollection {
 }
 
 export interface PostgresCollectionRecord extends PostgresCollection {
+  __name: tableNames.collections,
```

Comment:
This is by far the part of this approach that I like the least, but it's the only workable solution that I could find to enforce that a separate table name argument to a function matches the record type specified. And using enums to references the table names makes me feel slightly better about it.